torch/ao/quantization/_pt2e/prepare.py

```diff
@@ -0,0 +1,115 @@
+from ._quantizer import Quantizer
+from torch._subclasses import FakeTensor
+from torch.ao.quantization.prepare import (
+    _maybe_insert_input_observers_for_node,
+    _maybe_insert_output_observer_for_node,
+    _is_observer_in_same_graph,
+    _maybe_make_input_output_share_observers,
+    _remove_output_observer,
+    _maybe_insert_observers_before_graph_output,
+    _save_state
+)
+from torch.fx import GraphModule
+
+from torch.ao.quantization import QConfigMapping
+from torch.ao.quantization.custom_config import PrepareCustomConfig
+from typing import Dict, Tuple, Any
+
+def prepare(
+        model: GraphModule,
+        quantizer: Quantizer,
+        is_qat: bool,
+        node_name_to_scope: Dict[str, Tuple[str, type]],
+) -> GraphModule:
+    model = quantizer.annotate(model)
+    quantizer.validate(model)
+
+    # Since we are mutating the graph as we go, we iterate over the original
+    # nodes before observer insertion, instead of model.graph.nodes.
+    nodes_before_observation = list(model.graph.nodes)
+
+    for node in nodes_before_observation:
+
+        if node.op == "placeholder":
+            pass
+        elif node.op in ("call_module", "call_method", "call_function"):
+            this_node_dtype_info = node.meta["target_dtype_info"]
+            if "val" in node.meta:
+                output_is_a_tensor = (
+                    this_node_dtype_info is not None and
+                    isinstance(node.meta["val"], FakeTensor)
+                )
+            else:
+                output_is_a_tensor = this_node_dtype_info is not None
+
+            skip_inserting_observers = (
+                not output_is_a_tensor
+            )
+
+            if skip_inserting_observers:
+                continue
+
+            _maybe_insert_input_observers_for_node(
+                node,
+                None, # qconfig
+                model,
+                named_modules,
```

Comment:
What is the use of `named_module`?

Reply:
this is needed to store a mapping from observer name to observer instance right now: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/prepare.py#L663 maybe it can be improved later after refactoring